src/app/services/gallery-service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from "@angular/core";
+import {Http,Response} from "@angular/http";
+import { Observable} from "rxjs";
+import { map, filter } from "rxjs/operators";
+import { GlobalUrl } from "../constant/global-url";
+@Injectable()
+
+export class GalleryService{
+
+constructor(private http:Http){}
+ link=GlobalUrl.url;
+getAll() : Observable <any> {
+return this.http.get(this.link).
+pipe(map((res:Response)=>{
+    return res.json();
+}));
+}
+
+getByCategary(categary : string) : Observable <any>{
```

Comment:
use proper indentation for code. to better understand next time

Reply:
should i add commends for better identation of code